compiler/lib/Dialect/mhlo/Transforms/CanonicalizeExt.cpp

```diff
@@ -1377,6 +1377,49 @@ struct FoldLargeConcatenate : public OpRewritePattern<mhlo::ConcatenateOp> {
   }
 };
 
+struct CanonicalizeClamp : public OpRewritePattern<mhlo::ClampOp> {
+  using OpRewritePattern<mhlo::ClampOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(mhlo::ClampOp op,
+                                PatternRewriter &rewriter) const override {
+    auto minConst = op.getMin().getDefiningOp<mhlo::ConstantOp>();
+    auto maxConst = op.getMax().getDefiningOp<mhlo::ConstantOp>();
+    if (!minConst || !maxConst) {
+      return failure();
+    }
+    auto minAttr =
+        minConst.getValueAttr().dyn_cast_or_null<DenseElementsAttr>();
+    auto maxAttr =
+        maxConst.getValueAttr().dyn_cast_or_null<DenseElementsAttr>();
+    if (!minAttr || !maxAttr) {
+      return failure();
+    }
+    // replace splat const with scalar
+    if (minAttr.isSplat() && minAttr.getType().getRank() > 0 && //
+        maxAttr.isSplat() && maxAttr.getType().getRank() > 0) {
+      minAttr = minAttr.dyn_cast<SplatElementsAttr>().resizeSplat(
+          minAttr.getType().clone(SmallVector<int64_t>{}));
+      maxAttr = maxAttr.dyn_cast<SplatElementsAttr>().resizeSplat(
+          maxAttr.getType().clone(SmallVector<int64_t>{}));
+      rewriter.setInsertionPoint(minConst);
```

Comment:
Seems it's not necessary to `setInsertionPoint`. The default point of `OpRewritePattern`'s rewriter is before the matched op.